web/controllers/v1/bundle_dynamic_config_controller.ex

```diff
@@ -76,27 +76,34 @@ defmodule Cog.V1.BundleDynamicConfigController do
       nil ->
         not_found(conn, "Bundle #{bundle_id} not found")
       bundle ->
-        case Bundles.create_dynamic_config_for_bundle(bundle, params) do
-          {:ok, dyn_config} ->
-            location = case Map.get(params, "layer") do
-                         "base" ->
-                           Helpers.bundle_dynamic_config_path(conn, :show_layer,
-                                                              dyn_config.bundle_id,
-                                                              dyn_config.layer, %{})
-                         _ ->
-                           Helpers.bundle_dynamic_config_path(conn, :show_layer,
-                                                              dyn_config.bundle_id,
-                                                              dyn_config.layer,
-                                                              dyn_config.name, %{})
-                       end
-            conn
-            |> put_status(:created)
-            |> put_resp_header("location", location)
-            |> render("show.json", dynamic_config: dyn_config)
-          {:error, changeset} ->
+        try do
+          case Bundles.create_dynamic_config_for_bundle!(bundle, params) do
+            {:ok, dyn_config} ->
+              location = case Map.get(params, "layer") do
+                           "base" ->
+                             Helpers.bundle_dynamic_config_path(conn, :show_layer,
+                               dyn_config.bundle_id,
+                               dyn_config.layer, %{})
+                           _ ->
+                             Helpers.bundle_dynamic_config_path(conn, :show_layer,
+                               dyn_config.bundle_id,
+                               dyn_config.layer,
+                               dyn_config.name, %{})
+                         end
+              conn
+              |> put_status(:created)
+              |> put_resp_header("location", location)
+              |> render("show.json", dynamic_config: dyn_config)
+            {:error, changeset} ->
+              conn
+              |> put_status(:unprocessable_entity)
+              |> render(Cog.ChangesetView, "error.json", changeset: changeset)
+          end
+        rescue
```

Comment:
This is good, but I'm curious if Phoenix's built-in exception handling will handle this. If it can't out of the box, then I suspect there's just a small bit of configuration we'd need to set in order for it to work.

That being said, I'm certain this is not the only place in the API where we're doing this. It may be worth leaving this as is, and then following up to make a more principled pass through the rest of the code to fix things up.